server/src/test/java/org/elasticsearch/transport/ConnectionManagerTests.java

```diff
@@ -124,11 +124,10 @@ public void onNodeDisconnected(DiscoveryNode node, Transport.Connection connecti
         assertEquals(1, nodeDisconnectedCount.get());
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/49903")
     public void testConcurrentConnectsAndDisconnects() throws BrokenBarrierException, InterruptedException {
         DiscoveryNode node = new DiscoveryNode("", new TransportAddress(InetAddress.getLoopbackAddress(), 0), Version.CURRENT);
-        Transport.Connection connection = new TestConnect(node);
         doAnswer(invocationOnMock -> {
+            Transport.Connection connection = new TestConnect(node);
```

Comment:
Could we keep track of all of these connections and assert that all but one of them has been closed when the dust has settled?